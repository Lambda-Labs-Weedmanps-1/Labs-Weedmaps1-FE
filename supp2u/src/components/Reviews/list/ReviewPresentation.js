import React, {useEffect, useState} from 'react'
import Axios from 'axios'
import './reviews.sass'
import Reviews from "../../Reviews/add";
import ShowReview from "../../Reviews/show";


function ReviewPresentation(props) {
    //the GET call sets this state with the review data
    const [review, setReview] = useState([{
        "customer_id": null,
        "rating": null,
        "review": ""
    }]);
    let [hasReview, setHasReview] = useState(false);
    let [loading, setLoading] = useState(false);


    useEffect(() => {
        Axios.get(`${process.env.REACT_APP_BACKEND_URL}businesses/${props.business_id}/reviews`)
            .then(res => {
                setReview(res.data);

                // check if current signed in user have already left a review
                for (let i = 0; i < res.data.length; i++) {
                    if (res.data[i].customer_id === parseInt(localStorage.customer_id)) {
                        setHasReview(true);
                    }
                }
            })
            .catch(err => {
                console.log('ERROR POST\n', err)
            });
        setLoading(true)
    }, []);


    const routeReviewToEdit = (id) => {
        props.history.push(`/review/${id}`)
    };

    return (
        <div className={"review-list"}> {/* the class name is just for identifying the div on browser*/}
            {/* this renders reviews only if there ar reviews */}
            {review.customer_id === null ?
                <p>This restaurant has no reviews yet! <br/>Eaten there? Tell us how it went by adding a review!</p>
                :
                <>
                    <h1>Reviews</h1>
                    {/* this function maps all the reviews out for the restaurant */}
                    {review.map(review => {
                        return (
                            <ShowReview key={review} review={review} routeReviewToEdit={routeReviewToEdit}/>
                        )
                    })}
                </>
            }
            {/* check of user have left review already */}
            {hasReview ?
                <p> You have already left review, Thank you! </p>
                : loading ?
                    <Reviews business_id={props.business_id}/> : "loading"
            }
        </div>
    )
}

export default ReviewPresentation
