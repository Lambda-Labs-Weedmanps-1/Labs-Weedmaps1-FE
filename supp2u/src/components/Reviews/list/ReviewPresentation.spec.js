import React from 'react';
import { mount, configure, shallow, enzyme } from 'enzyme';
import { MemoryRouter } from 'react-router';
import Adapter from 'enzyme-adapter-react-16';
import { act } from "react-dom/test-utils";

import ReviewPresentation from './ReviewPresentation';

configure({ adapter: new Adapter() });

// comp should render even with no review data for restaurant
describe('Reviews Comp Should Render', () => {
    it('should render correctly', () => {
      const reviews = shallow(<ReviewPresentation />);
    
      expect(reviews).toMatchSnapshot();

    });
    
});
 
// Currently this is having a state/store issue that kills the tests, will
// work out to fix 
//
// test('valid path should not redirect to 404', () => {
//   const wrapper = mount(
//     <MemoryRouter initialEntries={[ '/business/1' ]}>
//       <ReviewPresentation/>
//     </MemoryRouter>
//   );
//   expect(wrapper.find(ReviewPresentation)).toHaveLength(1);
// //   wrapper.unmount();
// });