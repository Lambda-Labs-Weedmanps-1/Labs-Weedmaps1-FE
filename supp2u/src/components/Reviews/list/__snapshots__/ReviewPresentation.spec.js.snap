// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reviews Comp Should Render should render correctly 1`] = `
<div
  className="review-list"
>
   
  <h1>
    Reviews
  </h1>
  <_default
    key="[object Object]"
    review={
      Object {
        "customer_id": null,
        "rating": null,
        "review": "",
      }
    }
    routeReviewToEdit={[Function]}
  />
  loading
</div>
`;
